File: laravel-12-ddd-standard/dotnet-skeleton/src/Infrastructure/Repository/InMemoryObraRepository.cs
using System.Collections.Concurrent;
using System.Threading.Tasks;

namespace App.DotNet.Skeleton.Infrastructure.Repository
{
    using App.DotNet.Skeleton.Domain.Entities;
    using App.DotNet.Skeleton.Domain.Repositories;

    public sealed class InMemoryObraRepository : IObraRepository
    {
        private readonly ConcurrentDictionary<string, Obra> _storage = new();

        public Task SaveAsync(Obra obra)
        {
            _storage[obra.Id] = obra;
            return Task.CompletedTask;
        }

        public Task<Obra?> FindByIdAsync(string id)
        {
            _storage.TryGetValue(id, out var obra);
            return Task.FromResult(obra);
        }
    }
}

File: laravel-12-ddd-standard/examples/simple-service/src/Infrastructure/Repository/InMemoryObraRepository.php
<?php
declare(strict_types=1);

namespace App\Examples\SimpleService\Infrastructure\Repository;

use App\Examples\SimpleService\Domain\Entities\Obra;
use App\Examples\SimpleService\Domain\Repositories\ObraRepositoryInterface;

final class InMemoryObraRepository implements ObraRepositoryInterface
{
    /** @var array<string, array> */
    private array $storage = [];

    public function save(Obra $obra): void
    {
        $this->storage[$obra->id()] = $obra->toArray();
    }

    public function findById(string $id): ?Obra
    {
        if (!isset($this->storage[$id])) {
            return null;
        }

        $data = $this->storage[$id];

        return new Obra((string) $data['id'], (string) $data['titulo'], $data['descripcion'] ?? null);
    }
}

